src/lib/components/CopyCode/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Element that renders copyable code.
+ */
+
+import {BaseElement} from "../BaseElement";
+
+/**
+ * Renders code block that can easily be copied.
+ *
+ * @extends {BaseElement}
+ * @final
+ */
+class CopyCode extends BaseElement {
+  static get properties() {
+    return {
+      code: {type: String},
+    };
+  }
+
+  constructor() {
+    super();
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute("tabindex", "0");
+
+    if (this.firstChild) {
+      const copyButton = document.createElement("input");
+      copyButton.setAttribute("type", "button");
+      copyButton.setAttribute("tabindex", "0");
+      copyButton.className = "w-copy-code-button";
+
+      this.firstChild.prepend(copyButton);
+
+      this.addEventListener("click", (e) => {
```

Comment:
Could you do something like:

```
this.copyButton.addEventListener("click", this.onCopy)
```

I think stylistically that may be consistent with some of our other custom elements. We've tended to declare handlers with the `on*` prefix versus doing them inline.

Reply:
Note that you'll probably want to do:

```
this.onCopy = this.onCopy.bind(this);
```

in this element's constructor to properly scope that handler.